app/src/main/java/org/simple/clinic/navigation/v2/fragments/BaseScreen.kt

```diff
@@ -61,7 +72,11 @@ abstract class BaseScreen<K: ScreenKey, M: Parcelable, E, F>: Fragment() {
   open fun additionalEventSources(): List<EventSource<E>> = emptyList()
 
   override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-    return inflater.inflate(layoutResId(), container, false)
+    val view = inflater.inflate(layoutResId(), container, false)
+
+    _binding = bindView(view)
+
+    return view
```

Comment:
Instead of this could we have used the return value from `bindView` method? 🤔 . Something like this perhaps.

```kt
  abstract fun bindView(layoutInflater: LayoutInflater, container: ViewGroup?): B

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
    return bindView(inflater, container).root
  }
```

That way we don't have to provide the `LayoutResId` and do the binding inflate method in the classes that extend `BaseScreen`. WDYT?

Reply:
I am not entirely sure how that would work? I decided to follow the pattern that we use in our current screens where we call `Binding.bind(this)` which means we have to inflate the view and then instantiate the binding by providing it a reference to the view.

Reply:
It's slightly different for fragments, we can directly inflate from the binding and use `binding.root` for returning the view.

https://developer.android.com/topic/libraries/view-binding